enumerable_methods.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |item|
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    result = []
+    my_each { |item| result.push(item) if yield(item) }
+    result
+  end
+
+  def my_all?(args = nil)
```

Comment:
Your method is returning TRUE when applying it to a falsy element.
`[1,2,3, 'string', 'another', 'text'].my_all?(String)`

Your method is returning TRUE when applying it to a falsy element.
`['string', 'another', 'spacing'].my_all?(/ing/)`